### PR TITLE
Test particle scraping for EB in RZ.

### PR DESCRIPTION
**Pull request by Yin-YinjianZhao**
In this PR, an automated test will be added in which particles are moving into an embedded boundary and they should be removed.

### CODE REVIEW

File: Source/EmbeddedBoundary/DistanceToEB.H [hunk 1]
```diff
@@ -66,7 +66,7 @@ amrex::RealVect interp_normal (int i, int j, int k, const amrex::Real W[AMREX_SP
     normal[2] += phi(i  , j+1, k+1) * dxi[2] * W[0][0] * W[1][1];
     normal[2] -= phi(i+1, j+1, k  ) * dxi[2] * W[0][1] * W[1][1];
     normal[2] += phi(i+1, j+1, k+1) * dxi[2] * W[0][1] * W[1][1];
-#elif (defined WARPX_DIM_XZ)
+#elif (defined WARPX_DIM_XZ || defined WARPX_DIM_RZ)
```
Comment (RemiLehe) [inline Source/EmbeddedBoundary/DistanceToEB.H]:
```suggestion
#elif defined(WARPX_DIM_XZ) || defined(WARPX_DIM_RZ)
```

File: Source/EmbeddedBoundary/WarpXInitEB.cpp [hunk 2]
```diff
@@ -59,7 +59,7 @@ namespace {
         AMREX_GPU_HOST_DEVICE inline
         amrex::Real operator() (AMREX_D_DECL(amrex::Real x, amrex::Real y,
                                              amrex::Real z)) const noexcept {
-#if defined(WARPX_DIM_XZ) || defined(WARPX_DIM_RZ)
+#if (AMREX_SPACEDIM == 2)
```
Comment (RemiLehe) [inline Source/EmbeddedBoundary/WarpXInitEB.cpp]:
```suggestion
#if defined(WARPX_DIM_XZ) || defined(WARPX_DIM_RZ)
```

File: Source/Particles/Gather/ScalarFieldGather.H [hunk 3]
```diff
@@ -41,24 +41,31 @@ void compute_weights_nodal (const amrex::ParticleReal xp,
...
     W[1][0] = 1.0 - W[1][1];
     W[2][0] = 1.0 - W[2][1];
-#elif (defined WARPX_DIM_XZ)
-    amrex::Real x = (xp - plo[0]) * dxi[0];
-    amrex::Real z = (zp - plo[1]) * dxi[1];
 
-    i = static_cast<int>(amrex::Math::floor(x));
-    j = static_cast<int>(amrex::Math::floor(z));
-    k = 0;
+#elif (defined WARPX_DIM_XZ || defined WARPX_DIM_RZ)
```
Comment (RemiLehe) [inline Source/Particles/Gather/ScalarFieldGather.H]:
```suggestion
#elif defined(WARPX_DIM_XZ) || defined(WARPX_DIM_RZ)
```

File: Source/Particles/Gather/ScalarFieldGather.H [hunk 4]
```diff
@@ -84,14 +91,14 @@ amrex::Real interp_field_nodal (int i, int j, int k,
     value += scalar_field(i+1, j  , k+1) * W[0][1] * W[1][0] * W[2][1];
     value += scalar_field(i  , j+1, k+1) * W[0][0] * W[1][1] * W[2][1];
     value += scalar_field(i+1, j+1, k+1) * W[0][1] * W[1][1] * W[2][1];
-#elif (defined WARPX_DIM_XZ)
+#elif (defined WARPX_DIM_XZ || defined WARPX_DIM_RZ)
```
Comment (RemiLehe) [inline Source/Particles/Gather/ScalarFieldGather.H]:
```suggestion
#elif defined(WARPX_DIM_XZ) || defined(WARPX_DIM_RZ)
```

**Comment by Yin-YinjianZhao** [comment]
An automated test will be added, which indicates the correctness that particles are removed upon reaching the embedded boundary, as shown in this animation.
xy

**Review by RemiLehe — APPROVED** [review]

File: Examples/Tests/scraping/analysis_rz.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+import numpy as np
+import yt
+
+sys.path.insert(1, '../../../../warpx/Regression/Checksum/')
+import checksumAPI
+
+tolerance = 0
+
+last_fn = sys.argv[1]
+ds = yt.load( last_fn )
```
Comment (RemiLehe) [inline Examples/Tests/scraping/analysis_rz.py]:
```suggestion
fn = sys.argv[1]
ds = yt.load( fn )
```

**Pull request merged**